### ISSUE TITLE
Viewport transform is not rounded to integers which results in blurry nodes on some resolution

### ISSUE THREAD

**Issue by andsav** [xyflow/xyflow#3282]
### Describe the Bug

Hi team 👋 

It looks like the transform applied to `.react-flow__viewport` is not rounded to integers. This doesn't cause any problem on a high resolution screen like a retina display, but noticed this causes some elements to appear blurry on low resolutions (in particular custom nodes with overflow).

![image](https://github.com/wbkd/react-flow/assets/5506087/e8b3e4b6-bf8d-41ba-855a-432d2c9ee5d0)

Would it be possible to change this to round to the nearest integer when panning stops? As a workaround, I am currently using  the `onMove` and using `panBy` from the internal store to round to the nearest integer. 

### Steps to Reproduce the Bug or Issue

Pan the canvas and inspect the `.react-flow__viewport`, the transform is not always an integer. 

### Expected behavior

Would expect it to be an integer.

### Platform

- OS: MacOS
- Browser: Chrome
- Version: 11.7.4

**Comment by rsdouglas** [comment]
+1 this is causing issues for us as well

**Comment by aleib** [comment]
Have this problem as well 👍

**Comment by moklick** [comment]
easiest thing to do would be to have a controlled viewport. As a workaround we could offer a prop to control the rounding. I will think about this! Maybe we can add something for the next release.

**Comment by moklick** [comment]
Can you create an example that shows the issue? Is it just text that is blurry or what kind of elements?

**Comment by moklick** [comment]
I couldn't find a good way to introduce a workaround for this issue in v11. You need to wait for v12, which will have a controlled viewport that lets you format it the way you want.

**Comment by aleib** [comment]
> Can you create an example that shows the issue? Is it just text that is blurry or what kind of elements? Is this really about the x,y transform or also about the scale?

Hi @moklick, apologies for the late reply. We ended up solving our issue differently a few days ago. It turned out that rounding the translation of the canvas only solved our scenario sometimes, and only when at 100% zoom. This issue also only presented itself on Chrome for low res screens.

For us we found nested css `overflow` rules inside a node. Removing the duplicated rule removed the blur.

**Comment by Pronink** [comment]
I've spent several days trying to add styles that are supposed to fix this kind of blurry errors, such as 
```css
transform: translateZ(0);
transform: translate3d(0,0,0);
filter: blur(0);
overflow: hidden; /*etc*/
```
Ideas from https://stackoverflow.com/questions/15464055/css-transition-effect-makes-image-blurry-moves-image-1px-in-chrome and https://stackoverflow.com/questions/14677490/blurry-text-after-using-css-transform-scale-in-chrome?rq=3

-------------------------------
But none of them worked, even made it worse. Finally remembering when I started to experience this error, I realized that this error appeared when I implemented some animations to the blurred area of the node. I finally discovered that the style that caused the blurring was:
```css
animation-fill-mode: forwards
```

I hope my experience can help others!
